pkg/util/converter/prom.go

```diff
@@ -161,24 +162,28 @@ func readPrometheusData(iter *jsoniter.Iterator, opt Options) *backend.DataRespo
 }
 
 // will return strings or exemplars
-func readArrayData(iter *jsoniter.Iterator) *backend.DataResponse {
-	lookup := make(map[string]*data.Field)
+func readArrayData(iter *jsoniter.Iterator, opts Options) *backend.DataResponse {
```

Comment:
should the function name have exemplars in it?

Reply:
i'm bad at naming things, but in this case i think this might have been something you named originally 🙃. there's also `readLabelsOrExemplars`, so i'm not sure i have a better idea at the moment 🤔. definitely open to suggestions

this PR is more of a test of adding the sampling to the parser since it was trivial to port. https://github.com/grafana/grafana/pull/49858 is the PR that does it in a post-processing step.